testsrc/main/mondrian/rolap/aggmatcher/AggMeasureFactCountTest.java

```diff
@@ -0,0 +1,204 @@
+
+            + "</Table> \n"
+            + "<DimensionUsage name=\"Time\" source=\"Time\" foreignKey=\"time_id\"/> \n"
+            + "<Measure name=\"Unit Sales\" column=\"unit_sales\" aggregator=\"avg\"\n"
+            + "   formatString=\"Standard\"/>\n"
+            + "<Measure name=\"Store Cost\" column=\"store_cost\" aggregator=\"avg\"\n"
+            + "   formatString=\"#,###.00\"/>\n"
+            + "<Measure name=\"Store Sales\" column=\"store_sales\" aggregator=\"avg\"\n"
+            + "   formatString=\"#,###.00\"/>\n"
+            + "</Cube>\n"
+            + "</Schema>";
+
+    private final String QUERY = ""
+            + "select [Time].[Quarter].Members on columns, \n"
+            + "{[Measures].[Store Sales], [Measures].[Store Cost], [Measures].[Unit Sales]} on rows "
+            + "from [Sales]";
+
+    @Override
+    protected String getFileName() {
+        return "agg_measure_fact_count_test.csv";
+    }
+
+    @Override
+    public void setUp() throws Exception {
+        propSaver.set(propSaver.properties.GenerateFormattedSql, true);
+        propSaver.set(propSaver.properties.UseAggregates, true);
+        propSaver.set(propSaver.properties.ReadAggregates, true);
+        propSaver.set(propSaver.properties.IgnoreInvalidMembers, true);
+
+        super.setUp();
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Override
+    protected String getDirectoryName() {
+        return "testsrc/main/mondrian/rolap/aggmatcher";
+    }
+
+    @Override
+    protected String getCubeDescription() {
+        return "";
+    }
+
+    public void testDefaultRecognition() {
+        String schema = getAggSchema(null);
+
+        verifySameAggAndNot(QUERY, schema);
+    }
+
+    public void testAggName() {
+        String agg = ""
+                + "<AggName name=\"agg_c_6_fact_csv_2016\">\n"
+                + "    <AggFactCount column=\"FACT_COUNT\"/>\n"
+                + "    <AggMeasureFactCount column=\"store_sales_fact_count\" measure=\"store_sales\" />\n"
+                + "    <AggMeasureFactCount column=\"store_cost_fact_count\" measure=\"store_cost\" />\n"
+                + "    <AggMeasureFactCount column=\"unit_sales_fact_count\" measure=\"unit_sales\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Unit Sales]\" column=\"UNIT_SALES\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Store Cost]\" column=\"STORE_COST\" />\n"
+                + "    <AggMeasure name=\"[Measures].[Store Sales]\" column=\"STORE_SALES\" />\n"
+                + "    <AggLevel name=\"[Time].[Year]\" column=\"the_year\" />\n"
+                + "    <AggLevel name=\"[Time].[Quarter]\" column=\"quarter\" />\n"
+                + "    <AggLevel name=\"[Time].[Month]\" column=\"month_of_year\" />\n"
+                + "</AggName>\n";
+
+        String schema = getAggSchema(agg);
+
+        verifySameAggAndNot(QUERY, schema);
+    }
+
```

Comment:
I might be missing something, but these tests look identical to each other.
Also, they don't actually validate that an aggtable is used for the avg calculation.  You could verify that expected SQL is fired, instead.